subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,1930 @@
+                private String name;
+                private int age;
+
+                public String getName() {
+                    return name;
+                }
+
+                public void setName(String name) {
+                    this.name = name;
+                }
+
+                public int getAge() {
+                    return age;
+                }
+
+                public void setAge(int age) {
+                    this.age = age;
+                }
+            }
+        """
+
+        file('src/test/org/samplePersonTest.java') << """
+            package org.sample;
+
+            import org.apache.commons.beanutils.PropertyUtils;
+
+            public class PersonTest {
+                @Test
+                public void testPerson() {
+                    Object person = new Person();
+                    PropertyUtils.setSimpleProperty(person, "name", "Bart Simpson");
+                    PropertyUtils.setSimpleProperty(person, "age", 38);
+                    assertEquals("Bart Simpson", person.getName());
+                    assertEquals(38, person.getAge());
+                }
+            }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        desc                | dependencyNotation
+        'GAV string'        | gavStr(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'GAV map'           | gavMap(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'named args'        | namedArgs(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+    }
+
+    def "can add dependencies using a non-String CharSequence: #type"() {
+        given:
+        buildFile << """
+        import org.apache.commons.lang3.text.StrBuilder;
+
+        buildscript {
+            ${mavenCentralRepository()}
+
+            dependencies {
+                classpath("org.apache.commons:commons-lang3:3.11")
+            }
+        }
+
+        plugins {
+          id 'java-library'
+        }
+
+        ${mavenCentralRepository()}
+
+        $type buf = $creationNotation
+
+        testing {
+            suites {
+                test {
+                    dependencies {
+                        implementation(buf)
+                    }
+                }
+            }
+        }
+
+        tasks.register('checkConfiguration') {
+            dependsOn test
+            doLast {
+                def testCompileClasspathFileNames = configurations.testCompileClasspath.files*.name
+                assert testCompileClasspathFileNames.containsAll('commons-lang3-3.11.jar')
+            }
+        }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        type            | creationNotation
+        'StringBuffer'  | "new StringBuffer('org.apache.commons:commons-lang3:3.11')"
+        'StringBuilder' | "new StringBuilder('org.apache.commons:commons-lang3:3.11')"
+        'StrBuilder'    | "new StrBuilder('org.apache.commons:commons-lang3:3.11')"
```

Comment:
I wouldn't bother doing something like this.  StringBuilder/GString is good enough to show that it's not just a String

Reply:
OK